File: protected/controllers/AccountController.vb
Imports GeoPortal.GeoPortal.Models.BusinessLogic
Imports GeoPortal.GeoPortal.Models.Data

Namespace GeoPortal
    Public Class AccountController
        Inherits System.Web.Mvc.Controller
        Private debug As Boolean = True
        Private Db As New getDBConnections()
        ' GET: /Account



        Function LogOn(ByVal Username As String, ByVal password As String) As JsonResult


            Db.getDBConnection()
            Dim user As UserDetails = Db.getUser(Username, password)

            Return Me.Json(user)

        End Function

        <CompressFilter()>
        Function getMyDetails() As JsonResult
            If debug = True Then

                Dim db As New getDBConnections
                db.getDBConnection()
                db.getUser("rfry", "Jan1981")

            End If

            If Not Session("User") Is Nothing Then

                Dim user As UserDetails = Session("User")

                Db.getDBConnection()

                Dim myAC As myAccount = Db.getMyAccount(user)

                Return Json(myAC)
            Else
                Dim msg As New jsonMsg()
                msg.success = False
                msg.message = "Not logged In!"

                Return Json(msg)
            End If

        End Function
        <CompressFilter()>
        Function UpdateMyDetails(ByVal UserName As String, ByVal firstName As String, ByVal lastName As String, ByVal Email As String, ByVal Email2 As String, ByVal Bio As String, ByVal Institution As String, ByVal Telephone As String, ByVal Address As String) As JsonResult
            Dim result As New jsonMsg
            Dim newUserDetails As New UserDetails

            newUserDetails.FirstName = firstName
            newUserDetails.LastName = lastName
            newUserDetails.Email = Email
            newUserDetails.UserName = UserName

            Dim updateBio As New MyAccountDetails

            updateBio.Bio = Bio
            updateBio.Address = Address
            updateBio.Institution = Institution
            updateBio.Telephone = Telephone

            Dim update As New getDBConnections
            update.getDBConnection()
            If update.updateMyAccount(newUserDetails, updateBio, Session.Item("User").UID) Then
                result.message = "Your Account was successfully updated!"
                result.success = True

            Else
                result.message = "There was a problem updating your account - please try again!"
                result.success = False

            End If

            Return Json(result)

        End Function

        <CompressFilter()>
        Function ChangePassword(ByVal oldPassword As String, ByVal newPW1 As String) As JsonResult
            Dim user As UserDetails = Session("User")
            Dim db As New getDBConnections
            db.getDBConnection()
            Return Json(db.ChangePassword(user.UID, oldPassword, newPW1))

        End Function

        

   

    End Class
End Namespace

File: protected/controllers/AccountController.php
<?

class AccountController extends Controller {

//        $debug = True;
//        $Db = new getDBConnections();


    Function actionresetPW() {
        $Email = "";
        if(isset($_POST['txtEmail'])) {
            $Email = $_POST['txtEmail'];
        }
        $Username = "";
        if(isset($_POST['txtUsername'])) {
            $Username = $_POST['txtUsername'];
        }

        if ($Email && $Email != "") {

//            $findUserStr = "Select id, username from alphausersdetails where email='" . $Email . "';";
//            $idSet = DataAdapter::DefaultExecuteAndRead($findUserStr);

            $idResultObject = DataAdapter::DefaultPDOExecuteAndRead(
                "Select id, username from alphausersdetails where email=:email", array(":email" => $Email));

//            Log::toFile('ids ' . print_r($idResultObject->resultObject, true));

            if(sizeof($idResultObject->resultObject) == 1) {

                $ID = $idResultObject->resultObject[0]->id;
                $Username = $idResultObject->resultObject[0]->username;

                $newPW = Users::randomPassword();

                $Db = new getDBConnections();
                $msg = $Db->ChangePassword($ID, "", $newPW, false);

                sendEmail::SendPasswordReset($Email, $Username, $newPW);

                echo json_encode($msg);

            } Else {
                echo json_encode(false);
            }
        }
    }

    Function actionLogOn() {
        $Username = "";
        if(isset($_POST['Username'])) {
            $Username = $_POST['Username'];
        }
        $password = "";
        if(isset($_POST['password'])) {
            $password = $_POST['password'];
        }

        $Db = new getDBConnections();
        $user = $Db->getUser($Username, $password);


        echo json_encode($user);

    }

    Function actiongetMyDetails() {
        $db = new getDBConnections;
        $db->getDBConnection();
//        If (variables::$debug == True) {
//
//            $db->getUser("rfry", "Jan1981");
//
//        };

        If (! Yii::app()->session["User"] == null) {

            $user = Yii::app()->session["User"];

            $myAC = $db->getMyAccount($user);

            echo json_encode($myAC);
        } Else {
            $msg = new jsonMsg();
            $msg->success = False;
            $msg->message = "Not logged In!";

            echo json_encode($msg);
        }

    }

    Function actionUpdateMyDetails() {
        $UserName = "";
        if(isset($_POST['UserName'])) {
            $UserName = $_POST['UserName'];
        }
        $firstName = "";
        if(isset($_POST['FirstName'])) {
            $firstName = $_POST['FirstName'];
        }
        $lastName = "";
        if(isset($_POST['LastName'])) {
            $lastName= $_POST['LastName'];
        }
        $Email= "";
        if(isset($_POST['Email'])) {
            $Email= $_POST['Email'];
        }
        $Email2= "";
        if(isset($_POST['Email2'])) {
            $Email2 = $_POST['Email2'];
        }
        $Bio = "";
        if(isset($_POST['Bio'])) {
            $Bio = $_POST['Bio'];
        }
        $Institution = "";
        if(isset($_POST['Institution'])) {
            $Institution = $_POST['Institution'];
        }
        $Telephone = "";
        if(isset($_POST['Telephone'])) {
            $Telephone = $_POST['Telephone'];
        }
        $Address = "";
        if(isset($_POST['Address'])) {
            $Address = $_POST['Address'];
        }


        $result = new jsonMsg;
        $newUserDetails = new UserDetails();

        $newUserDetails->FirstName = $firstName;
        $newUserDetails->LastName = $lastName;
        $newUserDetails->Email = $Email;
        $newUserDetails->UserName = $UserName;

        $updateBio = new myAccountDetails();

        $updateBio->Bio = $Bio;
        $updateBio->Address = $Address;
        $updateBio->Institution = $Institution;
        $updateBio->Telephone = $Telephone;

        $update = new getDBConnections();
        $update->getDBConnection();
        If ($update->updateMyAccount($newUserDetails, $updateBio, Yii::app()->session["User"]->UID)) {
            $result->message = "Your Account was successfully updated!";
            $result->success = True;

        } Else {
            $result->message = "There was a problem updating your account - please try again!";
            $result->success = False;

        }

        echo json_encode($result);

    }

    Function actionChangePassword() {
        $oldPassword = "";
        if(isset($_POST['oldPassword'])) {
            $oldPassword = $_POST['oldPassword'];
        }
        $newPW1 = "";
        if(isset($_POST['newPW1'])) {
            $newPW1 = $_POST['newPW1'];
        }

        $user = Yii::app()->session["User"];
        $db = new getDBConnections;
        $db->getDBConnection();
        echo json_encode($db->ChangePassword($user->UID, $oldPassword, $newPW1));
    }




}